crates/workspace/src/workspace.rs

```diff
@@ -2663,7 +2663,7 @@ impl Workspace {
         if self
             .follower_states
             .values()
-            .all(|state| state.leader_id != state.leader_id)
+            .all(|state| state.leader_id != leader_id)
```

Comment:
Talked this over with @ConradIrwin and it seems that the comparison should be against the `leader_id` in the outer scope.